superset-frontend/src/views/CRUD/data/hooks.ts

```diff
@@ -75,11 +76,36 @@ export function useQueryPreviewState<D extends BaseQueryObject = any>({
   };
 }
 
-export const UseGetDatasetsList = (queryParams: string | undefined) =>
-  SupersetClient.get({
-    endpoint: `/api/v1/dataset/?q=${queryParams}`,
-  })
-    .then(({ json }) => json)
-    .catch(error =>
-      logging.error('There was an error fetching dataset', error),
-    );
+export const useGetDatasetsList = (queryParams: string | undefined) => {
+  const [datasets, setDatasets] = useState<DatasetObject[]>([]);
+
+  const getDatasetsList = () =>
+    SupersetClient.get({
+      endpoint: `/api/v1/dataset/?q=${queryParams}`,
+    })
+      .then(({ json }) => {
+        setDatasets(json?.result);
+      })
+      .catch(error =>
+        logging.error('There was an error fetching dataset', error),
+      );
+
+  const datasetNames = datasets?.map(dataset => dataset.table_name);
+
+  return { getDatasetsList, datasets, datasetNames };
+};
+
+export const useGetDatasetRelatedObjects = (id: string) => {
+  const [usageCount, setUsageCount] = useState(0);
+
+  const getDatasetRelatedObjects = () =>
+    SupersetClient.get({
+      endpoint: `/api/v1/dataset/${id}/related_objects`,
+    })
+      .then(({ json }) => {
+        setUsageCount(json?.charts.count);
+      })
+      .catch(error => logging.error(error));
```

Comment:
Not blocking, but do we know how we want errors to appear if these requests fail?  If they should be displayed to the user in toasts, we could include the `useToasts` hook and call `addDangerToast` here and in `getDatasetsList` above.

Reply:
#22610 changes all the errors to use toasts, once it's merged I will be bringing those changes into this PR.